server/src/main/resources/routes

```diff
@@ -3,7 +3,10 @@
 # ~~~~
 
 # API routes
--> /api/v1  de.frosner.broccoli.ApiV1Router
+-> /api/v1  de.frosner.broccoli.routes.ApiV1Router
+
+# Downloads
+-> /downloads/ de.frosner.broccoli.routes.DownloadsRouter
```

Comment:
What is the reasoning behind not making this part of our versioned API? IMHO downloading logs for an instance task allocation is not different from asking for its parameters. Also `/downloads` sounds like its a folder mapping where you can download static files. But it's actually something to refresh to get more logs.

@Gerrrr @lunaryorn what do you think?

Reply:
> What I'm saying is that it is not part of the UI. 

The UI uses it as of now.

If we moved it to the stable and versioned API we would have to provide more guarantees than we do currently.  We would need to introduce content negotiation to support JSON and plain text, we would have to specify parameter ranges for the log size, etc.

All of this is missing currently, because I treated it as a UI endpoint.

As an example of the implication of "stabilizing" this endpoint:  What happens if the user tries to load a 2GB log file?  Currently we allocate a string for the entire logfile (because that's simple), but since the UI only gets the last some KBs there's no harm.  If we stabilize the endpoint we need to refactor the code to stream the log file from Nomad to the frontend to avoid excessive memory allocation for large log files.  That's easy for plain text, but I don't know whether Play can encode large strings in a streaming fashion. 

If it can't we suddenly need to two code paths for different content types or impose a maximum limit on the log size, but then how to we pick this limit?  How to we make sure the limit cuts the log file correctly, ie doesn't split in the middle of surrogate pairs rendering the log unreadable to the client?

If we can provide satisfying answers to these questions we can make it part of the stable API.  If we can't we shouldn't, because—since it is a "stable" API—any bug we make becomes part of the interface and can't be fixed without introducing new API versions or potentially breaking clients.

> I don't know how this endpoint is going to evolve.

Then why plan for change?  

> I didn't get the last part with the curls. If the dedicated API router is there why doesn't it return 404 if the API route does not exist?

It does, but I think we're talking past each other.  I referred to why I introduced `/downloads` as a prefix instead of adding a top-level route, not about `/downloads/` versus `/api/`.

As I said I do not think that this endpoint is part of the API.

Reply:
Ok. To summarize we don't have an API endpoint to get logs at the moment, only a very ugly part of the UI. Correct?

Works for me! 👍

Reply:
@FRosner Yes that's the situation from my point of view.  From this state we can slowly refine to  get a nicer UI… an API for logs is less important to me as long as we don't have specific need.

Reply:
Agreed thank you!